UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs

```diff
@@ -11,15 +11,54 @@ namespace MLAgents
 {
     public static class GrpcExtensions
     {
+
+        /// <summary>
+        /// Converts a AgentInfo to a protobuf generated AgentInfoActionPairProto
+        /// </summary>
+        /// <returns>The protobuf version of the AgentInfoActionPairProto.</returns>
+        public static AgentInfoActionPairProto ToInfoActionPairProto(this AgentInfo ai)
+        {
+            var agentInfoProto = new AgentInfoProto
+            {
+                Reward = ai.reward,
+                MaxStepReached = ai.maxStepReached,
+                Done = ai.done,
+                Id = ai.id,
+            };
+
+            if (ai.actionMasks != null)
+            {
+                agentInfoProto.ActionMask.AddRange(ai.actionMasks);
+            }
+
+            if (ai.observations != null)
+            {
+                foreach (var obs in ai.observations)
+                {
+                    agentInfoProto.Observations.Add(obs.ToProto());
+                }
+            }
```

Comment:
Can't this block just be `ai.ToAgentInfoProto()`?

Reply:
Nice catch